codec/json.go

```diff
@@ -28,3 +34,22 @@ func ProtoMarshalJSON(msg proto.Message, resolver jsonpb.AnyResolver) ([]byte, e
 
 	return buf.Bytes(), nil
 }
+
+// ProtoMarshalJSONI same as ProtoMarshalJSON, but does msg type inspection to assert
+// that it implements `proto.Message` and return an error if it doesn't.
+func ProtoMarshalJSONI(msg interface{}, resolver jsonpb.AnyResolver) ([]byte, error) {
+	msgProto, err := AssertProtoMsg(msg)
+	if err != nil {
+		return nil, err
+	}
+	return ProtoMarshalJSON(msgProto, resolver)
+}
+
+// AssertProtoMsg casts i to a proto.Message. Returns an error if it's not possible.
+func AssertProtoMsg(i interface{}) (proto.Message, error) {
+	pm, ok := i.(proto.Message)
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "Expecting protobuf Message type, got %T", i)
+	}
+	return pm, nil
+}
```

Comment:
Please revert these functions. This is not desirable. Just do a type assertion inline.

Reply:
moved to internal